backend/src/api/middlewares/validation.ts

```diff
@@ -1,13 +1,12 @@
 import { NextFunction, Request, Response } from "express";
 import { getLogger } from "../../utils/logger";
-import Joi from "joi";
 
 const logger = getLogger();
 
 const validationMiddleware =
-  (schema: Joi.ObjectSchema, property: "body" | "query") =>
+  (schema: any, property: "body" | "query") =>
```

Comment:
u can type the schema as 

z.ZodObject<any>

so then u get the types to do the rest of the operations, u should be able to figure it out from there. if not let me know :)

Reply:
z.ZodObject<any> **

Reply:
Okay, let me know if I was doing it right for da new commit or nah😼

Reply:
you've got the right idea, but you need to update the methods that use the objects since they are typed with the older ones. in the interfaces file, delete the old ones and then update the services to reference the inferred type.

the best way i'd suggest doing this is by defining the type in the schema file too, export it, then import in the controller and services as well as anywhere else they are used.